src/user-post/user-post.module.ts

```diff
@@ -4,30 +4,35 @@ import {
   NestModule,
   RequestMethod,
 } from '@nestjs/common';
-import { PostController } from './post-controller/post-controller.controller';
-import { PostService } from './service/post/post.service';
+import { PostController } from './controller/post.controller';
+import { PostService } from './service/post.service';
 import { TypeOrmModule } from '@nestjs/typeorm';
-import { Post } from './entities/Post.entity';
+import { Posts } from './entities/Post.entity';
 import { PreauthMiddleware } from '../Middleware/auth/auth.guard';
 import { FirebaseApp } from '../Firebase/firebase.service';
+import { User } from '../authentication/entities/user.entity';
 
 @Module({
-  imports: [TypeOrmModule.forFeature([Post])],
+  imports: [TypeOrmModule.forFeature([Posts, User])],
   controllers: [PostController],
   providers: [PostService, FirebaseApp],
 })
 export class UserPostModule implements NestModule {
-  configure(consumer: MiddlewareConsumer): any {
+  configure(consumer: MiddlewareConsumer) {
     consumer
       .apply(PreauthMiddleware)
       .exclude(
         {
-          path: 'post/createPost',
-          method: RequestMethod.ALL,
+          path: '/post/getAllPosts',
+          method: RequestMethod.GET,
         },
         {
-          path: 'post/getPosts',
-          method: RequestMethod.ALL,
+          path: 'post/getAllPostsLength',
+          method: RequestMethod.GET,
+        },
+        {
+          path: 'post/getAllPosts/:length',
+          method: RequestMethod.GET,
         },
```

Comment:
Why why?

Reply:
I was previously needed that part to exclude paths that shouldn't be checked by our middleware. However, now that I'm using a Guard, that part is no longer necessary.